cpp/src/lance/arrow/fragment.cc

```diff
@@ -101,6 +111,10 @@ ::arrow::Result<std::vector<LanceFragment::FileReaderWithSchema>> LanceFragment:
 
 const std::shared_ptr<format::Schema>& LanceFragment::schema() const { return manifest_->schema(); }
 
+const std::shared_ptr<lance::format::DataFragment>& LanceFragment::data_fragment() const {
```

Comment:
i keep forgetting what is DataFragment vs LanceFragment

Reply:
`DataFragment` is the POD, which maps to what is serialized.

`LanceFragment` is subclass of `arrow::Fragment`, which offers the I/O functions, i.e., `ScanBatchAsync`, `CountRows` and etc.